flutter-studio/testSrc/com/android/tools/idea/tests/gui/framework/FlutterGuiTestRule.java

```diff
@@ -0,0 +1,351 @@
+
+import com.android.tools.idea.gradle.util.LocalProperties;
+import com.android.tools.idea.sdk.IdeSdks;
+import com.android.tools.idea.tests.gui.framework.fixture.FlutterFrameFixture;
+import com.android.tools.idea.tests.gui.framework.fixture.FlutterWelcomeFrameFixture;
+import com.android.tools.idea.tests.gui.framework.fixture.IdeFrameFixture;
+import com.android.tools.idea.tests.gui.framework.fixture.IdeaFrameFixture;
+import com.android.tools.idea.tests.gui.framework.matcher.Matchers;
+import com.google.common.collect.ImmutableList;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.io.FileUtil;
+import com.intellij.openapi.util.io.FileUtilRt;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.openapi.vfs.VfsUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.openapi.wm.impl.IdeFrameImpl;
+import org.fest.swing.core.Robot;
+import org.fest.swing.exception.WaitTimedOutError;
+import org.jdom.Document;
+import org.jdom.Element;
+import org.jdom.input.SAXBuilder;
+import org.jdom.xpath.XPath;
+import org.jetbrains.annotations.NotNull;
+import org.junit.AssumptionViolatedException;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.junit.runner.Description;
+import org.junit.runners.model.MultipleFailureException;
+import org.junit.runners.model.Statement;
+
+import javax.swing.*;
+import java.awt.*;
+import java.beans.PropertyChangeListener;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.truth.TruthJUnit.assume;
+import static org.fest.reflect.core.Reflection.*;
+
+@SuppressWarnings("Duplicates") // Adapted from com.android.tools.idea.tests.gui.framework.GuiTestRule
+public class FlutterGuiTestRule implements TestRule {
+
+  /**
+   * Hack to solve focus issue when running with no window manager
+   */
+  private static final boolean HAS_EXTERNAL_WINDOW_MANAGER = Toolkit.getDefaultToolkit().isFrameStateSupported(Frame.MAXIMIZED_BOTH);
+  private final RobotTestRule myRobotTestRule = new RobotTestRule();
+  private final LeakCheck myLeakCheck = new LeakCheck();
+  private final PropertyChangeListener myGlobalFocusListener = e -> {
+    Object oldValue = e.getOldValue();
+    if ("permanentFocusOwner".equals(e.getPropertyName()) && oldValue instanceof Component && e.getNewValue() == null) {
+      Window parentWindow = oldValue instanceof Window ? (Window)oldValue : SwingUtilities.getWindowAncestor((Component)oldValue);
+      if (parentWindow instanceof Dialog && ((Dialog)parentWindow).isModal()) {
+        Container parent = parentWindow.getParent();
+        if (parent != null && parent.isVisible()) {
+          System.out.println("Focus Listener: Request focus!");
```

Comment:
I see a few `System.out.println()`s in this PR; d owe still need them (or where they for development)? Should we switch to using Log.info() or similar instead?

Reply:
These are some of the rough spots that need to be cleaned up.